cycles-ledger/src/storage.rs

```diff
@@ -775,12 +775,17 @@ fn check_invariants(s: &State) {
     }
 }
 
+/// The maximum amount of bytes a 64-bit number can occupy when encoded in LEB128.
+const MAX_U64_ENCODING_BYTES: usize = 10;
+
 pub fn populate_last_block_hash_and_hash_tree(
     hash_tree: &mut RbTree<&'static str, Vec<u8>>,
     last_block_index: u64,
     last_block_hash: Hash,
 ) {
-    hash_tree.insert("last_block_index", last_block_index.to_be_bytes().to_vec());
+    let mut last_block_index_buf = Vec::with_capacity(MAX_U64_ENCODING_BYTES);
+    leb128::write::unsigned(&mut last_block_index_buf, last_block_index).unwrap();
```

Comment:
What happens if there is a panic here?

Reply:
I added the unit test to at least cover the minimum and maximum values for `u64`, so I don't see how this could fail (except if e.g., the `Vec::with_capacity()` call succeeds, but `leb128::write::unsigned` fails to write to the memory). If this were to fail, then either the upgrade (or install) would fail (if it happens in `State::compute_last_block_hash_and_hash_tree`), the transaction would fail (if the panic occurrs as part of the call to `State::emit_block`), or the call to `penalize` would fail (I'm not sure where this is called from).

Reply:
Thanks for looking into it! If we are confident that the `unwrap` is highly unlikely to fail, we can leave it as it is. I'm always a bit nervous when I see an `unwrap` without any comment outside of test code.   :slightly_smiling_face: